### DISCUSSION TITLE
Can I use ttl to rs232 or do I have to get a Max3232?

### DISCUSSION THREAD

**Discussion by bobcov** [softwarecrash/Solar2MQTT-ESP8266#87] [General]
I can order one, but I just wanted to use what I had here. Why doesn't this use the rx and tx pins to ttl to rs232?  I worked on another project where those pins were used. What is the advantage or disadvantage to using D7 and D6?

Also, if you have blinking two times as I do, putting aside for the moment that I am not hardware compliant in my setup, does that  mean no connection or does it mean no response was understood or there is a connection, but nothing replied?

I will go dig through the readme again, but I didn't see on first reading (but I know as soon as I post this, I will eat my words on second search) how to use the serial feature.  I am not sure if it is suppose to send to the inverter and then send responses to mqtt server or what exactly..

Thank you.

**Comment by derLoosi** [comment]
Hey,

you can find the blink codes in the Wiki

https://github.com/softwarecrash/Solar2MQTT/wiki/Blink-Codes

greetings
Daniel

**Reply by bobcov** [reply]
Hi, let me ask it another way: Does no connection mean the serial port as far as can be determined has nothing attached or can it also mean something is attached but it did not reply in a way understood by the software?  I ask because I have a Steca MPPT 5020 I am testing with until the Steca Solarix PLI is available. They both use similar codes so I was not sure if no connection could be protocol related or literally no serial communication. 

BTW, I placed an order. In the meantime, I will try to get this going.

**Comment by softwarecrash** [comment]
> I can order one, but I just wanted to use what I had here. Why doesn't this use the rx and tx pins to ttl to rs232? I worked on another project where those pins were used. What is the advantage or disadvantage to using D7 and D6?
> 

the reason is that RX/TX hardware Serial is, and reserved for debug.
ttl<->rs232 is max2323 (or clones)

**Reply by bobcov** [reply]
Ah, yes. Thanks for reminding me about the painful situation of trying to reflash and having to disconnect a pin to do that. So, my TTL to RS232 should have worked, in theory. I don't think it is acting correctly. I put a breakout connector on it and RXD was lit all the time and it seemed to be picking up noise when I moved a finger near the converter, TXD lit up. That's not a good sign.  I'm hopeful all will be fine with the board I ordered from you today.

Sorry to be a pain, but I can't find out what that serial setting does. Can you explain it or give me a link? Does it send commands to the inverter?

**Reply by all-solutions** [reply]
Just one question before processing the order: 
does your Steca MPPT 5020 work with the Watchpower or Solarpower PC software? Because if not, we can cancel the order immediately ;-)

It sends query commands and then receives the corresponding responses from the inverter, which it then processes and forwards via WebUI and MQTT.

**Reply by softwarecrash** [reply]
> Ah, yes. Thanks for reminding me about the painful situation of trying to reflash and having to disconnect a pin to do that. So, my TTL to RS232 should have worked, in theory. I don't think it is acting correctly. I put a breakout connector on it and RXD was lit all the time and it seemed to be picking up noise when I moved a finger near the converter, TXD lit up. That's not a good sign.  I'm hopeful all will be fine with the board I ordered from you today.
> 
> Sorry to be a pain, but I can't find out what that serial setting does. Can you explain it or give me a link? Does it send commands to the inverter?

What you mean with serial settings?
You can Control the Inverter over mqtt/webhook/serial console, Just Take a Look at the wiki

**Reply by bobcov** [reply]
I wasn't sure because I didn't get any reaction from the Steca when i tried to send commands either because it is unsupported or my hardware is not up to par. The inverter this is actually supposed to work on is the Steca Solarix PLI 5000-48 which I saw in the list of supported devices.  Because they use a very similar command set, I thought that the other Steca would work, but it either has a different protocol or something is wrong with my setup.

Watchpower didn't work for me because I never got it configured. The administrator password in the manual was not the right one and I didn't have time to play around, so I dumped it.

I'm going to take a look again at the protocol list for both of the Steca. I think they're pretty much the same which if true, proves that I've got bad hardware, bad technique, bad karma or some mix of all three.

I wanted to try the PLI 5000-48 today with the ESP. Charged up the batteries, but when I started to clip on the positive lead with a large alligator clip,  with the inverter power off, there was a sizeable spark which was unexpected. I took that as a sign to stop. I'm waiting for a reply from Steca as to why there would be such current draw  with nothing attached to the inverter. No short measured across the battery leads.

**Reply by softwarecrash** [reply]
you have to learn a lot....
the inverters without load hold a "bus" voltage around 350-400v with a pulse pump. so when the device is on it draw itemdaly 50-150W so that gives you a nice little spark.

so in clearance,
please carefull and think befor do things
the selled PCB are not directly releated to this project but have the license to do that
both software and hardware are is as "is" so when you burn down your house by do strange things, dont ask us for help ;)

**Reply by all-solutions** [reply]
And quite honestly: You order a PCBs where there is a mandatory field in the order processing that you confirm that your inverter works with Watchpower. And you don't even know it.
Legally, I am on the safe side with this query and the purchase is binding. However, I will cancel the order immediately and refund the money. If YOU are sure that your inverter works with Watchpower (or Solarpower), you are welcome to place a new order. Anything else would be a waste of money if you then want to return the purchase and I have to create a return slip for you for the return shipment (at my expense).

![image](https://github.com/softwarecrash/Solar2MQTT/assets/17761850/c160facf-0d20-4759-ae75-d13642ad14b3)

**Reply by bobcov** [reply]
Well, I can just check it again and order again.  But that seems kind of silly to me. This circumstance arises from conflicting directives. Your website says the inverter is ON THE LIST OF CONFIRMED WORKING INVERTERS. Therefore, is there any point to the Watchpower directive? If so, does it mean that an inverter can be on the confirmed working list without being Watchpower compatible? I think you have to agree you have directives which could be seen as clashing.

I've ordered again. Do not worry. I'm a supporter.  If it doesn't work, it doesn't work. You mention that there is a lot of time and effort here, so please, take my money and don't lose sleep over it. 

About the inverter's pulse bump, does this hold true with an inverter attached to nothing and with the power switch in the OFF position? The only point of the inverter being with me is to go through possible communication scenarios for mqtt, etc. It is not to have it connected to anything other than the batteries necessary to power it on.  I just don't get why it would do this with no load.  But if that's the way of the inverter world, as you said, I have a lot to learn.  And I don't know how much more careful I could be. I double checked polarity, I used 25mm cable, I had someone else check it as well. At some point, it has to be attached to power and if this is what it does, then being careful doesn't seem likely to change the outcome. Thanks for that explanation.  I'll see if Steca says anything before I try again. It just is not normal to me for that kind of reaction to occur.

**Reply by all-solutions** [reply]
Your inverter is NOT on the list. There is only one Steca on this list. That is the 48/5000 PLI. You have an MPPT 5020, which is NOT confirmed.
If, for example, Steca sources from eaSUN or handles the protocols like eaSUN, then one type is compatible, but the other is not.
Why? One is then PI30 and the other is MODBUS. Two completely different protocols.

**Reply by bobcov** [reply]
I've been patient, but now it is time for me to be a little more assertive. YOU need to learn to READ what the customer has written.  IF you could have managed that, you would have realized that I quite clearly stated that the inverter I will be using is a Steca Solarix PLI 5000-48 which is on the fill-in-the-blank list.  I have also stated that I am TRYING to be a supporter despite your very best efforts. Send the order. It will not be returned. I was going to support the project further, but you really do not make that easy.  Please leave me alone. You are not helpful.  Well, actually, you were helpful.  The modbus tip was great. Thank you for that. It helps me with other things involving the 5020...which again, is NOT the inverter which will be used.

**Reply by softwarecrash** [reply]
can you calm down please?
remember we do that for "nothing" the only trhing what we collect is the paper from the tax office...
we do that in free time as hobby....
the shop is only for official tax things.

**Reply by bobcov** [reply]
I've vented. I've made my frustration and source clear. I'm done. Thank you for this great project. I am sure by sunrise I will be an ardent fan and supporter.

**Comment by derLoosi** [comment]
Also, if i remember correctly, you are unable to flash the uC using usb when something is connected to the hardware rx/tx ports plus we had trouble with the connection to the inverter and it worked much better with softwareserial. But i might be wrong...

**Reply by softwarecrash** [reply]
thats correct

**Reply by bobcov** [reply]
What would it cost for you to make a hardware version with Ethernet output? I've seen a project to add Ethernet to the ESP8266 and I am wondering if that would be of interest and if so if I could afford it. I'd like to get MQTT via Ethernet without using WiFi bridges or similar.
Also, a small update. I know there was discussion that the MPPT 5020 is modbus. (This has no bearing on the Solarix PLI 5000-48, which is running fine for testing.) Both the Solarix and the MPPT respond to many, but not all of the same commands over RS232, but the MPPT returns a protocol inquiry of 72, which stops a lot of programs from talking with it even though it appears to be able to communicated with PI30.  Does this mean it is not modbus? I'm beginning to think a protocol number is a representation of a common command set found on a family of similar devices.

**Reply by softwarecrash** [reply]
> What would it cost for you to make a hardware version with Ethernet output? I've seen a project to add Ethernet to the ESP8266 and I am wondering if that would be of interest and if so if I could afford it. I'd like to get MQTT via Ethernet without using WiFi bridges or similar.

there is no plane to add LAN support to the firmware.

> Also, a small update. I know there was discussion that the MPPT 5020 is modbus. (This has no bearing on the Solarix PLI 5000-48, which is running fine for testing.) Both the Solarix and the MPPT respond to many, but not all of the same commands over RS232, but the MPPT returns a protocol inquiry of 72, which stops a lot of programs from talking with it even though it appears to be able to communicated with PI30. Does this mean it is not modbus? I'm beginning to think a protocol number is a representation of a common command set found on a family of similar devices.

i cant answer anything without a device or know the full communication protocol

**Reply by bobcov** [reply]
Comms are running fine with a higher quality Max3232 board. I think the chips I had before were counterfeit. Your board has not arrived yet, but its purpose will be to show what a professional implementation looks like. Hopefully, it will lead to additional sales.

**Reply by all-solutions** [reply]
The board is on its way. You have also received an e-mail with a tracking number. You can check where it is at any time.

**Reply by bobcov** [reply]
Gorgeous board, runs like a champ.  The only thing I am puzzled by is the webpage display format.  On initial setup, it looked neatly formatted on the page, but after setup, it's as if all of the formatting is gone and the fields for selection of settings, reset, etc, are all on one line.  I've checked on Chrome and Firefox on three different computers and they all display the same. I only mention this because it doesn't seem to me that this is by design and that I must have some setting wrong on all of my browsers. Oh, also same from the dongle you sent and the one I built myself. It's not critical, but it looks more professional nicely formatted on the page.

**Reply by softwarecrash** [reply]
what you mean with wrong fromat and all on one line? can you post a screenshot?

**Reply by bobcov** [reply]
I tried to paste images, but that didn't fly.  I see now there is an attach option which I will use now. First screen, with or without inverter attached looks normal.  2nd picture with or without looks not so normal.
![Formatted](https://github.com/softwarecrash/Solar2MQTT/assets/37782643/0d828e52-27ac-4d58-87c6-498be29587b1)
![Noformat](https://github.com/softwarecrash/Solar2MQTT/assets/37782643/28a2427f-9db4-46c7-be52-a6da417ebfb2)

**Reply by softwarecrash** [reply]
looks  wired....
can  you please open the page in firefox, hit F12 and then click on network like this:
![grafik](https://github.com/softwarecrash/Solar2MQTT/assets/44615614/6c41075e-a70c-43e7-89dd-4ef401deab02)

reload the page and post the data from the network analyze

**Reply by bobcov** [reply]
You know how it works when something is wrong and you go to the doctor and then it vanishes? Same thing here. I had not even gotten to F12. It had been unplugged. I plugged it back in and  refreshed the page and it came up normally in both Firefox and Chrome.  I then did the network analyze, but I didn't find a way to send it. Screenshot doesn't work on that part of the page and I cannot find a way to highlight all of the text. But it is working, so that's good enough for me. If it happens again, I will reboot it.

**Reply by softwarecrash** [reply]
basicly the thing what you see is when the webUI cant load the style definitions and script helpers, there are hostet on CDNs and not on the ESP itself and must loaded external,  that require a internet connection for the computer that view  the webUI, so when you as example have averry restrictive adblocker or something script blocking thing it looks like this.


the following adresses must be reachable by the device that view the  webUI:
```
https://cdn.jsdelivr.net/npm/bootstrap@5.3.1/dist/css/bootstrap.min.css
https://cdn.jsdelivr.net/npm/bootstrap-icons@1.3.0/font/bootstrap-icons.css
https://cdn.jsdelivr.net/npm/bootstrap@5.3.1/dist/js/bootstrap.bundle.min.js"
https://cdn.jsdelivr.net/npm/chart.js@4.3.0/dist/chart.umd.min.js "
https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js
```

**Reply by bobcov** [reply]
Okay, understood. This is good to know. Makes perfect sense. Thank zou.

**Reply by softwarecrash** [reply]
Haha... Just read why you have Problems

https://www.jsdelivr.com/blog/jsdelivr-may-outage-postmortem/

The js cdn was unavaible for short time, and Just in this time you have Seen the result of the problem

**Reply by bobcov** [reply]
I laughed too when I read this.  That's pretty funny. That's just my luck.  Glad it was figured out.  Thanks for the update.  Still chuckling...

**Comment by bobcov** [comment]
It is out for delivery tomorrow.  It's going to look so much nicer than what I have built.  Looking forward to it.

**Comment by bobcov** [comment]
Does anyone know the secret to sending data to Telegraf? I get parsing errors.  Anyone have a telegraf.conf file for Steca or Voltronic 5000-48 inverter? The data are never in the correct format no matter what parser I try. 
2024-05-10T10:18:59Z E! [inputs.mqtt_consumer] Error in plugin: metric parse error: expected tag at 1:2: "2"
2024-05-10T10:18:59Z E! [inputs.mqtt_consumer] Error in plugin: metric parse error: expected tag at 1:4: "230"
2024-05-10T10:18:59Z E! [inputs.mqtt_consumer] Error in plugin: metric parse error: expected tag at 1:5: "21.7"
2024-05-10T10:18:59Z E! [inputs.mqtt_consumer] Error in plugin: metric parse error: expected tag at 1:4: "230"
2024-05-10T10:18:59Z E! [inputs.mqtt_consumer] Error in plugin: metric parse error: expected tag at 1:3: "50"
2024-05-10T10:18:59Z E! [inputs.mqtt_consumer] Error in plugin: metric parse error: expected tag at 1:5: "21.7"
2024-05-10T10:18:59Z E! [inputs.mqtt_consumer] Error in plugin: metric parse error: expected tag at 1:5: "5000"
2024-05-10T10:18:59Z E! [inputs.mqtt_consumer] Error in plugin: metric parse error: expected tag at 1:5: "5000"
2024-05-10T10:18:59Z E! [inputs.mqtt_consumer] Error in plugin: metric parse error: expected tag at 1:3: "48"
2024-05-10T10:18:59Z E! [inputs.mqtt_consumer] Error in plugin: metric parse error: expected tag at 1:3: "46"

**Reply by softwarecrash** [reply]
You are in the wrong Project...

**Reply by bobcov** [reply]
Whoops!  I just assumed a lot of people used solar2mqtt to get data into telegraf, influx and so forth.  I thought somebody here may have done it using your device specifically.  Is this the wrong place to ask?

**Reply by softwarecrash** [reply]
for this question is better to make a new tread

**Reply by bobcov** [reply]
I posted a new thread. Also looking at some simulators so I can try to figure this out on my own.
Have a good weekend!
